app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -16,42 +28,59 @@ import { CaveatTypes } from '../../../../../shared/constants/permissions';
 import { UNKNOWN_TICKER_SYMBOL } from '../../../../../shared/constants/app';
 import { PermissionNames } from '../../../controllers/permissions';
 import { getValidUrl } from '../../util';
+import {
+  EndApprovalFlow,
+  GetCaveat,
+  GetChainPermissionsFeatureFlag,
+  RequestPermittedChainsPermission,
+  RequestUserApproval,
+  SetActiveNetwork,
+} from './types';
 
-export function findExistingNetwork(chainId, findNetworkConfigurationBy) {
+export function findExistingNetwork(
+  chainId: Hex,
+  findNetworkConfigurationBy: (options: { chainId: Hex }) => void,
+) {
   if (
     Object.values(BUILT_IN_INFURA_NETWORKS)
-      .map(({ chainId: id }) => id)
+      .map(({ chainId: id }) => id as Hex)
       .includes(chainId)
```

Comment:
We can define a type guard here to make the `as` assertions in the return statement unnecessary.

```suggestion
    ((inputId: Hex): inputId is keyof typeof CHAIN_ID_TO_TYPE_MAP =>
      Object.values(BUILT_IN_INFURA_NETWORKS)
        .map(({ chainId: id }) => id)
        .find((id) => id === inputId) !== undefined)(chainId)
```